### PR TITLE
"Rework of Moon/Stars/Planets plus WeatherEffect"

### PR DESCRIPTION
**Pull request by PitchBright**

### CODE REVIEW

File: src/main/java/stellarium/DrawSky.java [hunk 1]
```diff
@@ -295,8 +300,8 @@ else if (mc.theWorld.provider.isSurfaceWorld())
 				difm.set(VecMath.mult(sizem, difm));
 				difm2.set(VecMath.mult(sizem, difm2));
 
-				float alpha=Optics.getAlphaFromMagnitude(-17.0-StellarSky.getManager().Moon.mag-2.5*Math.log10(difactor),bglight);
-
+				float alpha=(float) (Optics.getAlphaFromMagnitude(-17.0-StellarSky.getManager().Moon.mag-2.5*Math.log10(difactor),bglight) / (StellarSky.getManager().Moon.getPhase()));
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Why divide by phase? This part is for moon luminance.

File: src/main/java/stellarium/DrawSky.java [hunk 2]
```diff
@@ -321,11 +326,12 @@ else if (mc.theWorld.provider.isSurfaceWorld())
 					double longdd=(double)longcd/(double)longn;
 					double latdd=1.0-(double)(latc+1)/(double)latn;
 
-					tessellator1.setColorRGBA_F(1.0f, 1.0f, 1.0f, (f4*moonilum[longc][latc]-0.15f*bglight)*2.0f);
+					float lightlevel = (0.875f*(bglight/2.1333334f));
+					tessellator1.setColorRGBA_F(1.0f - lightlevel, 1.0f - lightlevel, 1.0f - lightlevel, ((f4*moonilum[longc][latc]-0.015f*bglight)*2.0f));
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Why apply 1.0f - lightlevel?
and if you want this, just multiply this to the alpha value.

File: src/main/java/stellarium/DrawSky.java [hunk 3]
```diff
@@ -423,13 +430,10 @@ public void renderStar(float bglight, float weathereff, double time){
...
 					continue;
 
-//				float Turb = StellarSky.getManager().turb *(float) random.nextGaussian();
-//				Mag+=Turb;
-
 				if(VecMath.getZ(pos)<0) continue;
 
 				float size=0.5f;
-				float alpha=Optics.getAlphaFromMagnitudeSparkling(Mag, bglight);
+				float alpha=Optics.getAlphaFromMagnitudeSparkling(Mag, bglight) - (((1-weathereff)/1)*20f);
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Just no. Why divide it?

File: src/main/java/stellarium/DrawSky.java [hunk 4]
```diff
@@ -471,7 +475,7 @@ public void drawStellarObj(float bglight, float weathereff, EVector pos, double
 		if(VecMath.getZ(pos)<0) return;
 
 		float size=0.6f;
-		float alpha=Optics.getAlphaFromMagnitude(Mag, bglight);
+		float alpha=Optics.getAlphaFromMagnitude(Mag, bglight) - (((1-weathereff)/1)*20f);
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Same as above.

File: src/main/java/stellarium/DrawSky.java [hunk 5]
```diff
@@ -487,7 +491,7 @@ public void drawStellarObj(float bglight, float weathereff, EVector pos, double
 
 		tessellator1.startDrawingQuads();
 
-		tessellator1.setColorRGBA_F(1.0f, 1.0f, 1.0f, weathereff*alpha);
+		tessellator1.setColorRGBA_F(1.0f, 1.0f, 1.0f, alpha);
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Why remove weather effect here?

File: src/main/java/stellarium/DrawSky.java [hunk 6]
```diff
@@ -443,7 +447,7 @@ public void renderStar(float bglight, float weathereff, double time){
 
 				Color c=Color.getColor(B_V);
 
-				tessellator1.setColorRGBA(c.r, c.g, c.b, (int)(weathereff*alpha*255.0));
+				tessellator1.setColorRGBA(c.r, c.g, c.b, (int)(alpha*255.0));
```
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Why remove weather effect here?

File: src/main/java/stellarium/stellars/Moon.java [hunk 7]
```diff
@@ -120,7 +120,7 @@ public double illumination(EVector p){
 	
 	//Phase of the Moon(Update Needed)
 	public double getPhase(){
-		return (Math.PI-Math.acos(Spmath.getD(BOp.div(VecMath.dot(EcRPos, EcRPosG), BOp.mult(VecMath.size(EcRPos), VecMath.size(EcRPosG))))))/Math.PI;
+		return 1-(Math.PI-Math.acos(Spmath.getD(BOp.div(VecMath.dot(EcRPos, EcRPosG), BOp.mult(VecMath.size(EcRPos), VecMath.size(EcRPosG))))))/Math.PI;
```
Comment (Abastro) [inline src/main/java/stellarium/stellars/Moon.java]: Wrong formula. Check it again.

File: src/main/java/stellarium/DrawSky.java (hunk 1 above)
Comment (PitchBright) [inline src/main/java/stellarium/DrawSky.java]: This is for the Halo texture. A new moon should not have a halo, a full moon should have a full brightness halo, and a half moon's halo should be half-bright. I used the moon's phase to divide the alpha.

File: src/main/java/stellarium/DrawSky.java (hunk 2 above)
Comment (PitchBright) [inline src/main/java/stellarium/DrawSky.java]: It goes back to what I explained about Alpha not always being between 0 & 1. Alpha can be below 0, or above 1. In the case of the Moon, the alpha is extremely high, that's why it doesn't fade out during the day.

lightlevel is based on how much light there is from 0 to 1. That's why I use it instead.

File: src/main/java/stellarium/DrawSky.java (hunk 6 above)
Comment (PitchBright) [inline src/main/java/stellarium/DrawSky.java]: Because "alpha" is not always between 0 & 1.

File: src/main/java/stellarium/DrawSky.java (hunk 2 above)
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: Doesn't it come to the same result?

File: src/main/java/stellarium/DrawSky.java (hunk 6 above)
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: But it is filter which affects alpha in reality.

File: src/main/java/stellarium/DrawSky.java (hunk 1 above)
Comment (Abastro) [inline src/main/java/stellarium/DrawSky.java]: So why divide. It is no use, because new moon has less phase.

**Pull request merged**